src/include/error.h

```diff
@@ -174,6 +174,8 @@
 /*
  * EXTRA_DIAGNOSTICS_ENABLED --
  *  Fetch whether diagnostic asserts for the provided category are runtime enabled.
+ *  When HAVE_DIAGNOSTIC=1 WT_DIAG_ALL is always set on the connection and this function
```

Comment:
Np: Maybe we could rephrase to "When compiled with HAVE_DIAGNOSTIC=1, the WT_DIAG_ALL category is always set on the connection and this function will always return true for non-null sessions."

Reply:
👍 Fixed in 6ecc815674fb5f51755337d77be7f0e5776af658